theano/sandbox/gpuarray/opt.py

```diff
@@ -604,6 +607,34 @@ def local_gpua_gemm(node):
 
 
 @register_opt('fast_compile')
+@op_lifter([tensor.basic.Dot])
+def local_gpua_hgemm(node):
+    from theano.sandbox.cuda import nvcc_compiler
+    if nvcc_compiler.nvcc_version < '7.5':
+        return
```

Comment:
I would move that to just before the return and print a nice user warning that updating to cuda 7.5 would speed up his computation.